protocol/contracts/amo/test/RamosTestnetTempleTokenVault.sol

```diff
@@ -3,25 +3,37 @@ pragma solidity ^0.8.17;
 // Temple (amo/test/RamosTestnetTempleTokenVault.sol)
 
 import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
-import { IRamosProtocolTokenVault } from "contracts/interfaces/amo/helpers/IRamosProtocolTokenVault.sol";
+import { IRamosTokenVault } from "contracts/interfaces/amo/helpers/IRamosTokenVault.sol";
 import { ITempleERC20Token } from "contracts/interfaces/core/ITempleERC20Token.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
 /// @notice A version of the Protocol Token Vault which has mint/burn rights on the Temple ERC20
-contract RamosTestnetTempleTokenVault is IRamosProtocolTokenVault {
+contract RamosTestnetTempleTokenVault is IRamosTokenVault {
     using SafeERC20 for ITempleERC20Token;
+    using SafeERC20 for IERC20;
 
     ITempleERC20Token public immutable templeToken;
+    IERC20 public immutable quoteToken;
 
-    constructor (address _templeToken) {
+    constructor (address _templeToken, address _quoteToken) {
         templeToken = ITempleERC20Token(_templeToken);
+        quoteToken = IERC20(_quoteToken);
     }
 
-    function borrowProtocolToken(uint256 amount, address recipient) external {
+    function borrowProtocolToken(uint256 amount, address recipient ) external {
         templeToken.mint(recipient, amount);
     }
 
+    function borrowQuoteToken(uint256 amount, address recipient) external {
+        quoteToken.safeTransferFrom(address(this), recipient, amount);
+    }
```

Comment:
I understand that when a strategy borrows the quoteToken `e.g. Dai`, it also creates a templeDebtToken `dUsd` of the same value, so it can be repay laters, not sure if minting/burn dUsd should happen here?

Reply:
Yes, dUSD will be minted. The only thing allowed to mint/burn dUSD is the TRV though, so when you call trv.borrow() it will automatically be issued the dUSD